skmultiflow/classification/trees/hoeffding_adaptive_tree.py

```diff
@@ -0,0 +1,489 @@
+            if self._alternate_tree is not None:
+                self._alternate_tree.filter_instance_to_leaves(X, self, -999, update_splitter_counts, found_nodes)
+
+    class AdaLearningNode(LearningNodeNBAdaptive, NewNode):
+
+        def __init__(self, initial_class_observations):
+            LearningNodeNBAdaptive.__init__(self, initial_class_observations)
+            self.estimationErrorWeight = ADWIN()
+            self.ErrorChange = False
+            self.randomSeed = 1
+            self.classifierRandom = random.seed(self.randomSeed)
+
+        def calc_byte_size(self):
+            byte_size = self.__sizeof__()
+            if self.estimationErrorWeight is not None:
+                byte_size += self.estimationErrorWeight.get_length_estimation()
+            return byte_size
+
+        # Override NewNode
+        def number_leaves(self):
+            return 1
+
+        # Override NewNode
+        def get_error_estimation(self):
+            return self.estimationErrorWeight._estimation
+
+        # Override NewNode
+        def get_error_width(self):
+            return self.estimationErrorWeight._width
+
+        # Override NewNode
+        def is_null_error(self):
+            return (self.estimationErrorWeight is None)
+
+        def kill_tree_childs(self, hat):
+            pass
+
+        # Override NewNode
+        def learn_from_instance(self, X, y, weight, hat, parent, parent_branch):
+            true_class = y
+
+            k = np.random.poisson(1.0, self.classifierRandom)
+            if k > 0:
+                weight = weight * k
+
+            tmp = self.get_class_votes(X, hat)
+
+            class_prediction = max_index(tmp)
+
+            bl_correct = (true_class == class_prediction)
+
+            if self.estimationErrorWeight is None:
+                self.estimationErrorWeight = ADWIN()
+
+            old_error = self.get_error_estimation()
+
+            # Add element to Adwin
+            add = 0.0 if (bl_correct is True) else 1.0
+
+            self.estimationErrorWeight.add_element(add)
+            # Detect change with Adwin
+            self.ErrorChange = self.estimationErrorWeight.detected_change()
+
+            if (self.ErrorChange is True and old_error > self.get_error_estimation()):
+                self.ErrorChange = False
+
+            # Update statistics call LearningNodeNBAdaptive
+            super().learn_from_instance(X, y, weight, hat)  # CHECK changed self to super
+
+            # call ActiveLearningNode
+            weight_seen = self.get_weight_seen()
+
+            if weight_seen - self.get_weight_seen_at_last_split_evaluation() >= hat.grace_period:
+                hat._attempt_to_split(self, parent, parent_branch)
+                self.set_weight_seen_at_last_split_evaluation(weight_seen)
+
+        # Override LearningNodeNBAdaptive
+        def get_class_votes(self, X, ht):
+
+            dist = {}
+            prediction_option = ht.leaf_prediction
+
+            # TODO
```

Comment:
Why is this part commented out?

Reply:
Because we need to add the other ways of classifying the leaves, I used only NBAdaptive, for simplicity.

Reply:
Understood, since the code is stable we need to uncomment this code to test all options.

Reply:
Done!